Duo Security Connector

dicolanl opened the pull request:
Fixes #

## Proposed Changes
Includes:
Function Connector
Parsers
Detections
Workbook

shainw commented on Detections/DuoSecurity/IPEntity_DuoSecurity.yaml:
```diff
@@ -0,0 +1,56 @@
+  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+  | project LatestIndicatorTime, Description, ActivityGroupNames, IndicatorId, ThreatType, Url, ExpirationDateTime, ConfidenceScore, Duo_TimeGenerated,
+  TI_ipEntity, user_name_s, facto_s, result_s, application_name_s, event_type_s, txid_g, user_key_s, access_device_ip_s, access_device_location_city_s, access_device_location_state_s, access_device_location_country_s
+  | extend timestamp = Duo_TimeGenerated, IPCustomEntity = access_device_ip_s, AccountCustomEntity = user_name_s,
```
Drop the comma at the end of this line

dicolanl replied:
New commit is fixed

shainw commented on Detections/DuoSecurity/TrustMonitorEvent.yaml:
```diff
@@ -0,0 +1,30 @@
+
+  let timeframe = ago(5m);
+  DuoSecurityTrustMonitor_CL
+  | where TimeGenerated >= timeframe
```
Should there be a filter here for a certain type of event or is everything in this table something that should be alerted on?

dicolanl replied:
These are detections in Duo Security (https://duo.com/docs/trust-monitor) they do monitoring of their logs and generate alerts in Trust Monitor so they should be investigated in Sentinel

shainw reviewed (changes requested):
I don't see sample data set included which is required for parser testing.

dicolanl commented:
> I don't see sample data set included which is required for parser testing.

I have added via a new commit

dicolanl commented:
/azp run

preetikr commented:
@nazang / @v-jayakal - please review the data connector and workbook
@shainw for parser + detections.

v-maudan reviewed (changes requested):
@dicolanl Please update the following requested changes:

1. Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.
2. Please update changes related to Azure gov. Refer to this [PRLink ](https://github.com/Azure/Azure-Sentinel/pull/1762/files)for gov changes.

v-maudan commented on Parsers/DuoSecurity/DuoSecurityAdministrator.txt:
```diff
@@ -0,0 +1,30 @@
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as DuoSecurity<Log>
```
@dicolanl , Can you please let us know where we used these parsers.

v-maudan commented on Parsers/DuoSecurity/DuoSecurityAuthentication.txt:
```diff
@@ -0,0 +1,44 @@
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as DuoSecurity<Log>
```
Give proper function alias to save parser.[All parser]

dicolanl replied:
@v-maudan  do you have an example?

dicolanl replied:
@v-jayakal the parsers are not currently used in the workbooks or analytics.  They are available if the customer wants to use them

v-maudan commented on DataConnectors/DuoSecurity/DuoSecurity_API_FunctionApp.json:
```diff
@@ -0,0 +1,185 @@
+            "description": "**3. Configure the Function App**\n\n1. In the Function App, select the Function App Name and select **Configuration**.\n2. In the **Application settings** tab, select **+ New application setting**.\n3. Add each of the following seven (6) application settings individually, with their respective string values (case-sensitive): \n\t\tiKey\n\t\tsKey\n\t\tapiServer\n\t\tti ...
+        }
+    ]
+}
```
Please add a Metadata object within this file. See [https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json](https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json) for further detail.

dicolanl commented on DataConnectors/DuoSecurity/DuoSecurity_API_FunctionApp.json:
```diff
@@ -0,0 +1,185 @@
+{
+    "id": "DuoSecurity",
+    "title": "Duo Security (Preview)",
```
Done

dicolanl commented:
> @dicolanl Please update the following requested changes:
> 
> 1. Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.
> 2. Please update changes related to Azure gov. Refer to this [PRLink ](https://github.com/Azure/Azure-Sentinel/pull/1762/files)for gov changes.

I have moved to solutions folder and added support for Gov

v-maudan commented:
@dicolanl , Can you please fix PR validation errors. Thanks!

dicolanl commented:
> @dicolanl , Can you please fix PR validation errors. Thanks!

@v-maudan I created a new PR #2422 with CL schemas.  Need this approved and merged to fix validations!

dicolanl commented:
/azp run

dicolanl commented:
@v-maudan All checks now pass validation
